src/vellum/workflows/runner/runner.py

```diff
@@ -209,14 +210,18 @@ def __init__(
             self._entrypoints = self.workflow.get_entrypoints()
         elif trigger:
             # When trigger is provided, set up default state and filter entrypoints by trigger type
-            normalized_inputs = deepcopy(inputs) if inputs else self.workflow.get_default_inputs()
+            default_inputs = deepcopy(inputs) if inputs else None
             if state:
                 self._initial_state = deepcopy(state)
-                self._initial_state.meta.workflow_inputs = normalized_inputs
+                self._initial_state.meta.workflow_inputs = default_inputs
                 self._initial_state.meta.span_id = execution_id or uuid4()
                 self._initial_state.meta.workflow_definition = self.workflow.__class__
             else:
-                self._initial_state = self.workflow.get_default_state(normalized_inputs, execution_id)
+                self._initial_state = self.workflow.get_default_state(
+                    default_inputs,
+                    execution_id,
+                    fallback_to_default_inputs=False,
```

Comment:
[blocking] again, instead of the flag, we should pass in the default trigger attributes consistent with the default inputs. I believe this will make it easier to cutover fully to the latter when ready